src/deps.ml

```diff
@@ -8,15 +8,16 @@ type t =
 let paths t = t.paths
 
 let trace_path fn =
-  (Path.to_string fn, Utils.Cached_digest.file fn)
+  let digest, perms = Utils.Cached_digest.file fn in
+  (Path.to_string fn, digest, perms)
 
 let trace_var env var =
-  let value =
+  let value, present =
     match Env.get env var with
-    | None -> "unset"
-    | Some v -> Digest.string v |> Digest.to_hex
+    | None -> "unset", 0
+    | Some v -> Digest.string v |> Digest.to_hex, 1
```

Comment:
I don't think it's necessary to change that part. When a variable is unset, the trace value is "unset", otherwise it's the hash of the value. So the int part can be `0` in all cases. It's even preferable to store an option and return None in that case to avoid magic constants.

Reply:
Indeed, I misread the code.

Reply:
No problem, the value part can probably be rewritten as a digest option if you think that the current version is confusing.

Reply:
I was pondering whether a comment should be added.

Actually, what worries me is the prospect of someone
expecting to find a digest here (hence a string of a given
length), even though the values built by these function
are documented as *abstract*.

Reply:
Actually, why not make the values abstract?
Do you see any downside?

Reply:
I think it would be a good idea to make this abstract indeed!

Reply:
Done.